benches/bench.rs

```diff
@@ -0,0 +1,23 @@
+extern crate core;
+
+use criterion::{criterion_group, criterion_main, Criterion};
+use ff::Field;
+use pasta_curves::Fp;
+
+use spartan2::spartan::polys::eq::EqPolynomial;
+
+fn benchmarks_evaluate_incremental(c: &mut Criterion) {
+  let mut group = c.benchmark_group("evaluate_incremental");
+  (1..=20).step_by(2).for_each(|i| {
+    group.bench_with_input(format!("2^{}", i), &i, |b, &i| {
+      let eq_polynomial =
+        EqPolynomial::new(vec![Fp::random(&mut rand::thread_rng()); 2usize.pow(i)]);
```

Comment:
Also, beware the rng usage here. This code actually generates a random number `a` and repeats it `2usize.pow(i)]`-many times, instead of generating a fresh `a_i` for each entry in the vector.
You want to use `map` instead.

Reply:
Sure